File: ios/FintechSecurityModule.swift
import ExpoModulesCore

public final class FintechSecurityModule: Module {
  public func definition() -> ModuleDefinition {
    Name("FintechSecurity")

    AsyncFunction("getIdentifier") { () throws -> String in
      do {
        return try DeviceIdentifierProvider().getIdentifier()
      } catch is KeychainError {
        throw IdentifierStorageAccessException()
      }
    }
  }
}

File: android/src/main/java/expo/modules/fintechsecurity/FintechSecurityModule.kt
package expo.modules.fintechsecurity

import expo.modules.kotlin.exception.Exceptions
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class FintechSecurityModule : Module() {

  private val context
    get() = appContext.reactContext ?: throw Exceptions.ReactContextLost()

  override fun definition() = ModuleDefinition {
    Name("FintechSecurity")

    AsyncFunction("getIdentifier") {
      DeviceIdentifierProvider(context).getIdentifier()
    }
  }
}
